packages/zoe/src/contracts/loan/close.js

```diff
@@ -39,20 +42,22 @@ export const makeCloseLoanInvitation = (zcf, config) => {
     // Transfer the collateral to the repaySeat and remove the
     // required Loan amount. Any excess Loan amount is kept by the repaySeat.
     // Transfer the repaid loan amount to the lender
-
-    repaySeat.incrementBy(
-      collateralSeat.decrementBy(
-        harden({
-          Collateral: collateralSeat.getAmountAllocated(
-            'Collateral',
-            collateralBrand,
-          ),
-        }),
-      ),
+    atomicRearrange(
+      zcf,
+      harden([
+        [
+          collateralSeat,
+          repaySeat,
+          {
+            Collateral: collateralSeat.getAmountAllocated(
+              'Collateral',
+              collateralBrand,
+            ),
+          },
+        ],
+        [repaySeat, lenderSeat, { Loan: debt }],
+      ]),
```

Comment:
Would you mind pulling `collateralSeat.getAmountAllocated()` out? I think it makes the `atomicRearrange` much more readable.

```suggestion
    const collateralAmount = collateralSeat.getAmountAllocated(
      'Collateral',
      collateralBrand,
    );
    atomicRearrange(
      zcf,
      harden([
        [collateralSeat, repaySeat, { Collateral: collateralAmount }],
        [repaySeat, lenderSeat, { Loan: debt }],
      ]),
    );
```

Reply:
Much better! Done.